packages/safe-tools-client/app/controllers/schedule.ts

```diff
@@ -1,79 +1,58 @@
+import { SelectableToken } from '@cardstack/boxel/components/boxel/input/selectable-token';
+import { Web3Provider, getTokenBalancesForSafe } from '@cardstack/cardpay-sdk';
 import ApplicationController from '@cardstack/safe-tools-client/controllers/application';
 import NetworkService from '@cardstack/safe-tools-client/services/network';
+import { Safe } from '@cardstack/safe-tools-client/services/safes';
 import WalletService from '@cardstack/safe-tools-client/services/wallet';
-
 import Controller, { inject as controller } from '@ember/controller';
 import { action } from '@ember/object';
-import { default as Owner } from '@ember/owner';
+
 import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
 
+import { trackedFunction } from 'ember-resources/util/function';
+
+import TokensService from '../../services/tokens';
+
 import '../css/schedule.css';
 
 export default class Schedule extends Controller {
   @controller declare application: ApplicationController;
   @service declare wallet: WalletService;
   @service declare network: NetworkService;
+  @service declare tokens: TokensService;
+  @service('safes') declare safeService: SafesService;
 
   @tracked isDepositModalOpen = false;
-  @tracked isInitializingSafes = false;
+  @tracked selectedSafe?: Safe;
 
-  constructor(owner: Owner) {
-    super(owner);
-    this.initializeSafes();
-  }
+  tokenBalances = trackedFunction(this, async () => {
+    if (!this.safe) return;
 
-  @action async initializeSafes() {
-    try {
-      this.isInitializingSafes = true;
+    const ethersProvider = new Web3Provider(this.wallet.web3.currentProvider);
+    const tokenAddresses = this.tokens.transactionTokens.map(
+      (t: SelectableToken) => t.address
+    );
 
-      // TODO: replace with ember resources
-      // Waits a bit to get connection on refresh
-      setTimeout(async () => {
-        if (this.wallet.isConnected) {
-          await this.wallet.fetchSafes();
-        }
-        this.isInitializingSafes = false;
-      }, 500);
-    } catch (e) {
-      console.log('Error initializing safes', e);
-    }
+    return await getTokenBalancesForSafe(
```

Comment:
Just realized we'll need native token balance too. I'll use this SDK method: https://github.com/cardstack/cardstack/blob/main/packages/cardpay-sdk/sdk/assets.ts#L15